pkg/ddl/create_table.go

```diff
@@ -683,6 +683,12 @@ func BuildTableInfoWithStmt(ctx sessionctx.Context, s *ast.CreateTableStmt, dbCh
 		return nil, errors.Trace(err)
 	}
 
+	// set default shard row id bits and pre-split regions for table.
+	if !tbInfo.HasClusteredIndex() && tbInfo.TempTableType == model.TempTableNone {
```

Comment:
If user manually set these two vars in create table or alter table statements, I think we should use user's define values. what do you think?

Reply:
`handleTableOptions` is called on line 692, so it will overwrite the system variable setup here. Manually setup will always have a higher priority.